pkg/schedulingtypes/replicascheduler.go

```diff
@@ -108,6 +108,38 @@ func (s *ReplicaScheduler) Kind() string {
 	return RSPKind
 }
 
+func (s *ReplicaScheduler) RegisterPlugins(kind string, apiResource *metav1.APIResource, stopChan <-chan struct{}) {
+	var adapter adapters.Adapter
+	switch kind {
+	case FederatedDeployment:
+		adapter = adapters.NewFederatedDeploymentAdapter(s.fedClient)
+	case FederatedReplicaSet:
+		adapter = adapters.NewFederatedReplicaSetAdapter(s.fedClient)
+	default:
+		return
```

Comment:
I think returning an error is suggested if the kind is not recognized, so that the caller knows they are attempting to do something that is not supported.